src/clients/Client.cc

```diff
@@ -923,7 +923,9 @@ Client::handledEarlyAdaptationAbort()
 void
 Client::handleAdaptationBlocked(const Adaptation::Answer &answer)
 {
-    debugs(11,5, answer.ruleId);
+    const auto blockedAnswer = answer.blockedToChecklistAnswer();
+
+    debugs(11,5, blockedAnswer.lastCheckDescription());
 
     if (abortOnBadEntry("entry went bad while ICAP aborted"))
```

Comment:
Currently Answer::Block() is called only by Ecap::XactionRep::blockVirgin(), so I think this method cannot be initiated by ICAP...

Reply:
Please ignore this (possibly temporary) functionality difference between eCAP and ICAP as well as "ICAP" in the abortOnBadEntry() debugging text. The text should say "adaptation transaction" instead of "ICAP", but we should not fix this unless we need to modify this line for other reasons.